omegaconf/basecontainer.py

```diff
@@ -374,10 +374,11 @@ def _set_item_impl(self, key: Any, value: Any) -> None:
         input_config = isinstance(value, Container)
         target_node_ref = self._get_node(key)
         special_value = value is None or value == "???"
-        should_set_value = (
-            target_node_ref is not None
-            and isinstance(target_node_ref, Container)
-            and special_value
+        should_set_value = target_node_ref is not None and (
+            (
+                isinstance(target_node_ref, Container)
+                and (special_value or target_node_ref._metadata.ref_type)
+            )
```

Comment:
okay, this is simpler.
Can you articulate the logic behind using the ref_type of the node as an indication of the need to set vs wrap?

Reply:
Since listconfigs didn't use ref_type before, and should_set_value was only True when the target was a container and the value was 'special', checking if the container has a ref_type means that it should be changed instead of wrapping a new node.

Reply:
The fact ListConfig did not use ref_type before should not be a factor here.
Can this be simplified? reading your explanation here does not give me a solid "ahhh, I get it" feeling.